logic/peers.js

```diff
@@ -75,11 +75,13 @@ Peers.prototype.get = function (peer) {
 Peers.prototype.upsert = function (peer, insertOnly) {
 	// Insert new peer
 	var insert = function (peer) {
-		peer.updated = Date.now();
-		__private.peers[peer.string] = peer;
-
-		library.logger.debug('Inserted new peer', peer.string);
-		library.logger.trace('Inserted new peer', {peer: peer});
+		if (!_.isEmpty(modules.peers.acceptable([peer]))) {
+			peer.updated = Date.now();
+			__private.peers[peer.string] = peer;
+			library.logger.debug('Inserted new peer', peer.string);
+		} else {
+			library.logger.debug('Inserting peer rejected - peer is itself or has private address: ', peer.string);
```

Comment:
I think simply the following would be better: `library.logger.debug('Rejecting unacceptable peer', peer.string);`